File: backend/question-service/src/main/java/br/com/questionarium/question_service/dto/QuestionDTO.java
package br.com.questionarium.question_service.dto;

import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import br.com.questionarium.question_service.types.QuestionAccessLevel;
import br.com.questionarium.question_service.types.QuestionEducationLevel;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuestionDTO {
    private Long id;
    private Boolean multipleChoice;
    private Integer numberLines;
    private QuestionEducationLevel educationLevel;
    private Long userId;
    private String header;
    private String headerImage;
    private Long answerId;
    private Boolean enable;
    private QuestionAccessLevel accessLevel;
    private List<Long> tagIds;
    private List<AlternativeDTO> alternatives;
    private LocalDateTime creationDateTime;
    private LocalDateTime updateDateTime;
}

File: frontend/src/app/types/dto/QuestionDTO.ts
export interface QuestionDTO {
    id: number | null;
    multipleChoice: boolean;
    numberLines: number;
    educationLevel: number;
    userId: number;
    header: string;
    header_image?: string | null;
    answerId: number | null;
    enable: boolean;
    accessLevel: number;
    tagIds: number[] | null;
    creationDateTime: Date;
    updateDateTime: Date;
    // alternatives: Alternative[];
}


/*
  ENSINO_FUNDAMENTAL(0),
    ENSINO_MEDIO(1),
    ENSINO_SUPERIOR(2);

*/